core/src/main/scripts/importer/validateData.py

```diff
@@ -3421,7 +3423,32 @@ class TreatmentWiseFileValidator(MultipleDataFileValidator, metaclass=ABCMeta):
     prior_validated_header = None
     REQUIRED_HEADERS = ['treatment_id']
     OPTIONAL_HEADERS = ['name', 'description', 'url']
-    UNIQUE_COLUMNS = ['treatment_id']
+    UNIQUE_COLUMNS = ['treatment_id','name']
+
+    def __init__(self, *args, **kwargs):
+        super(TreatmentWiseFileValidator, self).__init__(*args, **kwargs)
+
+    def parseFeatureColumns(self, nonsample_col_vals):
+        self.checkDifferentNameInDb(nonsample_col_vals)
+        return MultipleDataFileValidator.parseFeatureColumns(self, nonsample_col_vals)
+    
+    def checkDifferentNameInDb(self, nonsample_col_vals):
+        # Check for different combinations of treatment_id and treatment_name
+        # in the database. If true, raise warnings for each discrepancy.
+        nonsample_cols =  self.nonsample_cols
+        if (nonsample_cols.index("name") != None):
+
+            treatmentId = nonsample_col_vals[ nonsample_cols.index("treatment_id") ]
+            fileTreatmentName = nonsample_col_vals[ nonsample_cols.index("name") ]
+            dbTreatmentName = self.portal.treatment_map[treatmentId]["name"]
```

Comment:
You are right. The code did not account for the fact that a particular treatment_id was not added to the database before. Please not that if a treatment_id is entered there is _always_ a name field added; this can not be a reason for the `dbTreatmentName` to be `None`. The new code is:

>            # check whether a name for the treatment has been
>            # registered in the database
>            dbTreatmentName = None
>            if self.portal.treatment_map[treatmentId] != None:
>                dbTreatmentName = self.portal.treatment_map[treatmentId]["name"]
>
>            # when a name has been registered for this treatment and 
>            # is different from the new name, issue a warning.
>            if dbTreatmentName != None and dbTreatmentName != fileTreatmentName:
>                    self.logger.warning(